Jenkinsfile

```diff
@@ -0,0 +1,139 @@
+                withDockerRegistry([ credentialsId: "dockerhub_${DOCKER_NAMESPACE}", url: ""]) {
+                    sh 'docker push relay-webapp:${BRANCH_NAME}'
+                }
+            }
+        }
+
+        stage('Delivery semver docker images') {
+            when { 
+                expression {
+                    return isVersionTag(readCurrentTag())
+                }
+            }
+            environment {
+                DOCKER_NAMESPACE = 'dopplerdock'
+                SEMVER_MAYOR = "${env.BRANCH_NAME.tokenize('.')[0]}"
+                SEMVER_MAYOR_MINOR = "${SEMVER_MAYOR}.${env.BRANCH_NAME.tokenize('.')[1]}"
+                SEMVER_MAYOR_MINOR_PATCH = "${SEMVER_MAYOR_MINOR}.${env.BRANCH_NAME.tokenize('.')[2]}"
+            }
+            steps {
+                sh 'docker tag relay-webapp:${BRANCH_NAME}-commit-${GIT_COMMIT} relay-webapp:${SEMVER_MAYOR}'
+                sh 'docker tag relay-webapp:${BRANCH_NAME}-commit-${GIT_COMMIT} relay-webapp:${SEMVER_MAYOR_MINOR}'
+                sh 'docker tag relay-webapp:${BRANCH_NAME}-commit-${GIT_COMMIT} relay-webapp:${SEMVER_MAYOR_MINOR_PATCH}'
+                sh 'docker tag relay-webapp:${BRANCH_NAME}-commit-${GIT_COMMIT} relay-webapp:latest'
+                
+                withDockerRegistry([ credentialsId: "dockerhub_${DOCKER_NAMESPACE}", url: ""]) {
+                    sh 'docker push relay-webapp:${SEMVER_MAYOR}'
+                    sh 'docker push relay-webapp:${SEMVER_MAYOR_MINOR}'
+                    sh 'docker push relay-webapp:${SEMVER_MAYOR_MINOR_PATCH}'
+                    sh 'docker push relay-webapp:latest'
+                }
+            }
+        }
+    }
+    post { 
+        cleanup { 
+            cleanWs()
+            dir("${env.WORKSPACE}@tmp") {
+                deleteDir()
+            }
+        }
+    }
+}
+
+def String defineEnvironmentByBranchName() {
+    switch(BRANCH_NAME) {
+    case ["main", "master", readCurrentTag()]:
+        environment = "PROD"
+        break
+    case "qa":
+        environment = "QA"
+        break
+    case "int":
+        environment = "INT"
+        break
+    default:
+        environment = "development"
+        break
+    }
+    echo "Environment: ${environment}"
+    return "${environment}"
+}
+
+def boolean isVersionTag(String tag) {
+    echo "checking version tag $tag"
+
+    if (tag == null) {
+        return false
+    }
+
+    // use your preferred pattern
+    def tagMatcher = tag =~ /v\d+\.\d+\.\d+/
+
+    return tagMatcher.matches()
+}
+
+// https://stackoverflow.com/questions/56030364/buildingtag-always-returns-false
+// workaround https://issues.jenkins-ci.org/browse/JENKINS-55987
+// TODO: read this value from Jenkins provided metadata
+def String readCurrentTag() {
+    return sh(returnStdout: true, script: "git describe --tags --match v?*.?*.?* --abbrev=0 --exact-match || echo ''").trim()
+}
```

Comment:
Now we are using a simpler version that [seems to work better](https://stackoverflow.com/questions/56030364/buildingtag-always-returns-false#answer-65483072):

https://github.com/FromDoppler/hello-microservice/blob/92e58b85074afda427cba2ba7ca5e98a5c3c9740/Jenkinsfile#L102-L107